fcrepo-clients/src/test/java/edu/unc/lib/dl/event/RepositoryPremisLoggerIT.java

```diff
@@ -166,4 +181,221 @@ public void getEventsModelForObjectWithoutLog() throws Exception {
 
         assertTrue(eventsModel.isEmpty());
     }
+
+    @Test
+    public void makeMultipleChangesSimultaneously() throws InterruptedException {
+        parentObject = repoObjFactory.createDepositRecord(null);
+        initPremisLogger(parentObject);
+
+        // make sure that there are no events in premis log before the writes
+        PremisLogger retrieveLogger = new RepositoryPremisLogger(parentObject, mockSession,
+                pidMinter, repoObjLoader, repoObjFactory);
+        Model initialLogModel = retrieveLogger.getEventsModel();
+        assertFalse("New premis already contains events", initialLogModel.listObjects().hasNext());
+
+        // add original event
+        Resource originalEventResc = logger.buildEvent(Premis.note)
+                .addEventDetail("first premis event")
+                .write();
+
+        // add new events
+        List<Thread> threads = new ArrayList<>();
+        List<Resource> events = new ArrayList<>();
+        List<String> eventUris = new ArrayList<>();
+
+        Resource event1Resc = logger.buildEvent(Premis.VirusCheck)
+                .addSoftwareAgent(SoftwareAgent.clamav.toString())
+                .create();
+        events.add(event1Resc);
+
+        Date ingestDate = Date.from(Instant.parse("2010-01-02T12:00:00Z"));
+        Resource event2Resc = logger.buildEvent(null, Premis.Ingestion, ingestDate)
+                .addEventDetail("Ingested")
+                .create();
+        events.add(event2Resc);
+
+        for (int i = 1; i <= 200; i++) {
+            Resource anotherEvent = logger.buildEvent(new PID("event" + System.currentTimeMillis() + i), Premis.note,
```

Comment:
At some point PID is going to become a interface, but this should be fine for now.